agent/command/git.go

```diff
@@ -166,24 +178,14 @@ func getProjectMethodAndToken(ctx context.Context, comm client.Communicator, td
 	globalToken := conf.Expansions.Get(evergreen.GlobalGitHubTokenExpansion)
 	token, err := parseToken(globalToken)
 	if err != nil {
-		return evergreen.CloneMethodLegacySSH, "", err
+		return "", "", err
 	}
 
-	switch conf.GetCloneMethod() {
-	// No clone method specified is equivalent to using legacy SSH.
-	case "", evergreen.CloneMethodLegacySSH:
-		return evergreen.CloneMethodLegacySSH, token, nil
-	case evergreen.CloneMethodOAuth:
-		if token == "" {
-			return evergreen.CloneMethodLegacySSH, "", errors.New("cannot clone using OAuth if explicit token from parameter and global token are both empty")
-		}
-		token, err := parseToken(globalToken)
-		return evergreen.CloneMethodOAuth, token, err
-	case evergreen.CloneMethodAccessToken:
-		return evergreen.CloneMethodLegacySSH, "", errors.New("cannot specify clone method access token")
+	if token == "" {
+		return "", "", errors.New("cannot clone using OAuth if explicit token from parameter and global token are both empty")
 	}
 
-	return "", "", errors.Errorf("unrecognized clone method '%s'", conf.GetCloneMethod())
+	return cloneMethodOAuth, token, err
```

Comment:
nit: `err` has already been checked on L180, so it might be clearer that this is the success case to just return nil error.